libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -795,6 +808,39 @@ public void onClick(DialogInterface dialog, int id) {
         dialog.show();
     }
 
+    private void showFeaturedImageConfirmationDialog(final int mediaId) {
+        if (mFeaturedImageId == 0) {
+            featuredImageSet(mediaId);
+        } else if (mFeaturedImageId != mediaId) {
+            Builder builder = new MaterialAlertDialogBuilder(getActivity());
+            builder.setTitle(R.string.replace_current_title);
+            builder.setMessage(R.string.replace_current_description);
+            builder.setPositiveButton(R.string.replace_current_confirmation,
+                    new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int id) {
+                            dialog.dismiss();
+                            featuredImageSet(mediaId);
+                        }
+                    });
+
+            builder.setNegativeButton(R.string.replace_current_cancel, new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int id) {
+                    dialog.dismiss();
+                }
+            });
+
+            AlertDialog dialog = builder.create();
+            dialog.show();
+        }
+    }
+
+    private void featuredImageSet(int mediaId) {
+        mEditorFragmentListener.updateFeaturedImage(mediaId, true);
```

Comment:
Looking at [the code that uses the `imagePicked` parameter](https://github.com/wordpress-mobile/WordPress-Android/blob/9f9a9c9d3b024debc62bf5b3211691d6390b1059/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java#L1002-L1008) (passed as `true` here), my impression is that it uses that param just to trigger a Tracks event during the media picking flow.

During the set-feature-image flow from inside an image block, I'd expect to pass `false` so to avoid triggering that event so, I'm not sure why the `true` here @SiobhyB , can you elaborate?

This also seems related to the concern you expressed in [the comment](https://github.com/wordpress-mobile/WordPress-Android/pull/14052#issuecomment-801029547).